core/kernel/dt_driver.c

```diff
@@ -677,12 +677,10 @@ static TEE_Result probe_dt_drivers_early(void)
 	TEE_Result res = TEE_ERROR_GENERIC;
 	const void *fdt = NULL;
 
-	if (!IS_ENABLED(CFG_EMBED_DTB))
+	fdt = get_dt();
+	if (!fdt)
 		return TEE_SUCCESS;
 
-	fdt = get_embedded_dt();
```

Comment:
May I ask a question？

In some case, fdt will be "external_dt.blob"，but in old codes that fdt will be NULL;
For example: if get_embedded_dt() return NULL, get_dt() maybe return "external_dt.blob".

I don’t' know what the impact will be.

Reply:
I believe this is addressed in @etienne-lms's proposal.

There is slight logical change still.

Previously if embedded device tree usage was enabled but it was not actually present then there would be assertion failure. Now this continues the boot (and may fail later on).

I would leave this to firmware integrator to make sure that they have configured things properly.